pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -135,10 +135,14 @@ public IndexLoadingConfig(InstanceDataManagerConfig instanceDataManagerConfig, T
     this(instanceDataManagerConfig, tableConfig, null);
   }
 
+  @VisibleForTesting
   public IndexLoadingConfig(TableConfig tableConfig, @Nullable Schema schema) {
     extractFromTableConfigAndSchema(tableConfig, schema);
   }
 
+  /**
+   * NOTE: Can be used in production code when we want to load a segment as is without any modifications.
```

Comment:
would this conflict with the case when user indeed wants to 'remove' all indexes?

Reply:
When table config and schema are not set, pre-processor is skipped